src/library_html5_webgpu.js

```diff
@@ -1,11 +1,96 @@
-mergeInto(LibraryManager.library, {
+{{{ (function() {
+  // Helper functions for code generation
+  global.html5_gpu = {
+    makeImportExport: function(snake_case, CamelCase) {
+      var s = '';
+      s += 'LibraryHTML5WebGPU.emscripten_webgpu_import_' + snake_case + '__sig = "ii";';
+      s += 'LibraryHTML5WebGPU.emscripten_webgpu_import_' + snake_case + '__deps = ["$WebGPU", "$JsValStore"];';
+      s += 'LibraryHTML5WebGPU.emscripten_webgpu_import_' + snake_case + ' = function(handle) { '
+      s += 'return WebGPU["mgr' + CamelCase + '"].create(JsValStore.get(handle));'
+      s += '};';
+
+      s += 'LibraryHTML5WebGPU.emscripten_webgpu_export_' + snake_case + '__sig = "ii";';
+      s += 'LibraryHTML5WebGPU.emscripten_webgpu_export_' + snake_case + '__deps = ["$WebGPU", "$JsValStore"];';
+      s += 'LibraryHTML5WebGPU.emscripten_webgpu_export_' + snake_case + ' = function(handle) { '
+      s += 'return JsValStore.add(WebGPU["mgr' + CamelCase + '"].get(handle));'
+      s += '};';
+      return s;
+    },
+  };
+  return null;
+})(); }}}
+
+
+var LibraryHTML5WebGPU = {
+  $JsValStore: {
+    values: {},
+    next_id: 1,
+
+    add: function (js_val) {
+      var id;
+      do {
+        id = JsValStore.next_id++;
+        if (JsValStore.next_id > 2147483647) JsValStore.next_id = 1; // Wraparound signed int32.
+      } while (id in JsValStore.values);
+
+      JsValStore.values[id] = js_val;
+      return id;
+    },
+    remove: function (id) {
+#if ASSERTIONS
+      assert(id in JsValStore.values);
+#endif
+      delete JsValStore.values[id];
+    },
+    get: function (id) {
+#if ASSERTIONS
+      assert(id === 0 || id in JsValStore.values);
+#endif
+      return JsValStore.values[id];
+    },
+  },
+
+  emscripten_release_js_handle__deps: ['$JsValStore'],
+  emscripten_release_js_handle: function (id) {
```

Comment:
This would be good to be named `emscripten_webgpu_release_js_handle` or something like that?

Reply:
Done